### PR TITLE
refactor: use core ResourceTable in cli

### PR DESCRIPTION
**Pull request by bartlomieju**
Follow up to #3134, #3136 and #3150

### CODE REVIEW

**Comment by bartlomieju** [comment]
@ry this is minimal landable patch with only `Repl` factored out. PTAL while I start working on breaking out child and worker resources

**Comment by bartlomieju** [comment]
Interestingly resource are now not properly dropped 🤔 investigating...

File: core/resources.rs [hunk 1]
```diff
@@ -23,24 +21,27 @@ type ResourceMap = HashMap<ResourceId, Box<dyn Resource>>;
...
 
-  pub fn get_mut<T: Resource>(
-    &mut self,
-    rid: ResourceId,
-  ) -> Result<&mut T, Error> {
-    let resource = self.map.get_mut(&rid).ok_or_else(bad_resource)?;
-    let resource = resource.downcast_mut::<T>().ok_or_else(bad_resource)?;
-    Ok(resource)
+  pub fn get_mut<T: Resource>(&mut self, rid: ResourceId) -> Option<&mut T> {
+    if let Some(resource) = self.map.get_mut(&rid) {
```
Comment (ry) [inline core/resources.rs]: Might as well make get_mut in core return a Option<&mut T> too?

Comment (bartlomieju) [reply core/resources.rs]: Not sure I understand. That's what this function does

Comment (ry) [reply core/resources.rs]: Err sorry! Misunderstanding on my part.

**Pull request merged**